diff --git a/node_modules/@cosmjs/proto-signing/.DS_Store b/node_modules/@cosmjs/proto-signing/.DS_Store
new file mode 100644
index 0000000..236b6f3
Binary files /dev/null and b/node_modules/@cosmjs/proto-signing/.DS_Store differ
diff --git a/node_modules/@cosmjs/proto-signing/build/pubkey.js b/node_modules/@cosmjs/proto-signing/build/pubkey.js
index 5562560..abe1d4b 100644
--- a/node_modules/@cosmjs/proto-signing/build/pubkey.js
+++ b/node_modules/@cosmjs/proto-signing/build/pubkey.js
@@ -25,7 +25,15 @@ function encodePubkey(pubkey) {
             value: Uint8Array.from(keys_3.PubKey.encode(pubkeyProto).finish()),
         });
     }
-    else if ((0, amino_1.isEd25519Pubkey)(pubkey)) {
+    else if ((0, amino_1.isEthSecp256k1Pubkey)(pubkey)) {
+        const pubkeyProto = keys_3.PubKey.fromPartial({
+            key: (0, encoding_1.fromBase64)(pubkey.value),
+        });
+        return any_1.Any.fromPartial({
+            typeUrl: "/cysicmint.crypto.v1.ethsecp256k1.PubKey",
+            value: Uint8Array.from(keys_3.PubKey.encode(pubkeyProto).finish()),
+        });
+    } else if ((0, amino_1.isEd25519Pubkey)(pubkey)) {
         const pubkeyProto = keys_1.PubKey.fromPartial({
             key: (0, encoding_1.fromBase64)(pubkey.value),
         });
@@ -56,11 +64,16 @@ exports.encodePubkey = encodePubkey;
  * In most cases you probably want to use `decodePubkey`.
  */
 function anyToSinglePubkey(pubkey) {
+    console.log('amino_1', amino_1)
     switch (pubkey.typeUrl) {
         case "/cosmos.crypto.secp256k1.PubKey": {
             const { key } = keys_3.PubKey.decode(pubkey.value);
             return (0, amino_1.encodeSecp256k1Pubkey)(key);
         }
+        case "/cysicmint.crypto.v1.ethsecp256k1.PubKey": {
+            const { key } = keys_3.PubKey.decode(pubkey.value);
+            return (0, amino_1.encodeEthSecp256k1Pubkey)(key);
+        }
         case "/cosmos.crypto.ed25519.PubKey": {
             const { key } = keys_1.PubKey.decode(pubkey.value);
             return (0, amino_1.encodeEd25519Pubkey)(key);
@@ -78,6 +91,7 @@ exports.anyToSinglePubkey = anyToSinglePubkey;
 function decodePubkey(pubkey) {
     switch (pubkey.typeUrl) {
         case "/cosmos.crypto.secp256k1.PubKey":
+        case "/cysicmint.crypto.v1.ethsecp256k1.PubKey":
         case "/cosmos.crypto.ed25519.PubKey": {
             return anyToSinglePubkey(pubkey);
         }